src/rgw/rgw_rest_s3.cc

```diff
@@ -2560,8 +2564,16 @@ int RGW_Auth_S3_Keystone_ValidateToken::validate_s3token(const string& auth_id,
     keystone_url.append("/");
   keystone_url.append("v2.0/s3tokens");
 
+  /* get authentication token for Keystone. */
+  string admin_token_id;
+  int r = RGWSwift::get_keystone_admin_token(cct, admin_token_id);
+  if (r < 0) {
+    dout(2) << "s3 keystone: cannot get token for keystone access" << dendl;
```

Comment:
@rzarzynski one small nitpick, please change this dout(2) to ldout(cct, 2). Once that's done I'll merge it.